digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/VerificationDao.scala

```diff
@@ -264,24 +267,36 @@ class VerificationDao {
   }
 
   def getModifiedAssetTypes(linkIds : Set[Long]) : List[LatestModificationInfo] = {
-    val modifiedAssetTypes = MassQuery.withIds(linkIds) { idTableName =>
-      sql"""
-           select assetTypeId, modifiedBy, modifiedDate
-           from (
-              select a.asset_type_id as assetTypeId, a.modified_by as modifiedBy, max(TO_DATE(TO_CHAR(a.modified_date, 'YYYY-MM-DD'), 'YYYY-MM-DD hh24:mi:ss')) as modifiedDate
+    val modifiedAssetTypes =
+      if (linkIds.size > 15000) {
+        logger.info(s"BATCH LOG Using mass query (${linkIds.size} links)")
+        MassQuery.withIds(linkIds) { idTableName =>
+          queryModifiedAssetTypes(s"select id from $idTableName")
+        }
+      } else {
+        logger.info(s"BATCH LOG Querying with id list (${linkIds.size} links)")
+        queryModifiedAssetTypes(s"${linkIds.mkString(",")}")
+      }
+    modifiedAssetTypes.map { case (assetTypeCode, modifiedBy, modifiedDate) =>
+      LatestModificationInfo(assetTypeCode,  modifiedBy, modifiedDate)
+    }
+  }
+
+  def queryModifiedAssetTypes(idQuery: String): List[(Int, Option[String], Option[DateTime])] = {
+    LogUtils.time(logger, "BATCH LOG Query modified asset types") {
+      sql"""select assetTypeId, modifiedBy, date(modifiedDate)
+            from (
+              select a.asset_type_id as assetTypeId, a.modified_by as modifiedBy, max(a.modified_date) as modifiedDate
               from asset a
               join asset_link al on a.id = al.asset_id
               join lrm_position lrm on lrm.id = al.position_id
               where a.modified_date is not null
               and a.valid_to is null
               and a.modified_by not in ('#${AutoGeneratedValues.allAutoGeneratedValues.mkString("','")}')
-              and lrm.link_id in (select id from #$idTableName)
```

Comment:
https://blog.crunchydata.com/blog/joins-or-subquery-in-postgresql-lessons-learned ihan kiintoisa